blockchain/blockdao/blockdao.go

```diff
@@ -295,6 +297,14 @@ func (dao *blockDAO) PutBlock(ctx context.Context, blk *block.Block) error {
 	lruCachePut(dao.headerCache, header.HashHeader(), &header)
 	timer.End()
 
+	g := genesis.MustExtractGenesisContext(ctx)
+	ctx = protocol.WithBlockCtx(ctx, protocol.BlockCtx{
+		BlockHeight:    blk.Height(),
+		BlockTimeStamp: blk.Timestamp(),
+		Producer:       blk.PublicKey().Address(),
+		GasLimit:       g.BlockGasLimitByHeight(blk.Height()),
+		BlobBaseFee:    *block.CalcBlobFee(blk.ExcessBlobGas()),
+	})
```

Comment:
why the blockctx was not needed?

Reply:
Previously, blockCtx was set inside the factory, but now, due to the introduction of BlobBaseFee requiring the previous block for initialization, and since the factory does not have this information internally, the setting has been moved ahead here in blockdao.